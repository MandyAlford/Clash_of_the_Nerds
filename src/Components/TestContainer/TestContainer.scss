.test-question {
  background-color: rgba(255, 255, 255, 0.9);
  padding: 20px;
  border: 5px solid #662324;
  float: left;
  margin: 50px;
  width: 500px;
  font-size: 30px;
  height: 400px;
}

.trivia-category {
  color: #FFF;
  font-size: 55px;
  margin-top: 50px;
  margin-right: 60%;
}

.trivia-header {
  color: #662324;
  margin-top: 50px;
}

.trivia-difficulty {
  color: #FFF;
  font-size: 25px;
  margin-right: 60%;
  margin-top: 10px;
}
