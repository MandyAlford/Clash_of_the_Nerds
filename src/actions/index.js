export const addUser = (user) => ({
    type: 'ADD_USER',
    user
})

export const addTrivia = (trivia) => ({
    type: 'ADD_TRIVIA',
    trivia
})

export const addCorrectGuess = (guess) => ({
  type: 'ADD_CORRECT_GUESS',
  guess
})

export const addIncorrectGuess = (guess) => ({
  type: 'ADD_INCORRECT_GUESS',
  guess
})

export const resetAnswers = () => ({
  type: 'RESET_STORE_ANSWERS'
})

export const saveUsersTrivia = (trivia) => ({
  type: 'SAVE_TRIVIA',
  trivia
})
