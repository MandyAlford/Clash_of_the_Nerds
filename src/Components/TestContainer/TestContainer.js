import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import Results from '../Results/Results.js';
import TriviaCard from '../TriviaCard/TriviaCard';
import './TestContainer.scss';
import { addTrivia } from '../../actions';
import { connect } from 'react-redux';
let currentQuestion, triviaData;
let questions;
let i = 0;

class TestContainer extends Component {
  constructor() {
    super()
    this.state={
      currentQuestion: [],
      redirect: false
    }
  }
  componentDidMount = () => {
    i = 0;
  }
  changeCard = (triviaData) => {
    if (i < 9 ) {
      console.log('first', i);
      i++
      this.setState({ chosenAnswer: [...this.state.currentQuestion, triviaData] })
      // return triviaData = <TriviaCard trivia={questions[i]} changeCard={this.changeCard} />
    } else {
      console.log('second', i);
      return null
      // i = 9;
      // return i
    }
  }
  formatQuestion = () => {
    questions = [];
    this.props.trivia.trivia.forEach(question => {
      if (question.question.includes('&#039;')) {
        let newQuestionObj = { ...question, question: question.question.replace('&#039;', '\'').replace('&#039;', '\'') }
        questions.push(newQuestionObj)
      } else if (question.question.includes('&amp;')) {
        let newQuestionObj = { ...question, question: question.question.replace('&amp;', '&') }
        questions.push(newQuestionObj)
      } else if (question.question.includes('&quot;')) {
        let newQuestionObj = { ...question, question: question.question.replace('&quot;', '\"').replace('&quot;', '\"') }
        questions.push(newQuestionObj)
      } else if (question.question.includes('&rsquo;s')) {
        let newQuestionObj = { ...question, question: question.question.replace('&rsquo;s', '\'s') }
        questions.push(newQuestionObj)
      } else {
        questions.push(question)
      }
    })
    return questions;
  }
   render() {
     this.formatQuestion();
     const triviaData = <TriviaCard trivia={questions[i]} changeCard={this.changeCard} />
     return (
       <section>
          <h1 className='trivia-category'>{questions[i].category}</h1>
          <h2 className='trivia-difficulty'>Difficulty: {questions[i].difficulty}</h2>
          <article className='test-question'>
          <h1 className='trivia-header'>Question: {i + 1}</h1>
          <h1 className='trivia-question'>{questions[i].question}</h1>
         </article>
         <article className='test-choice'>
          <div>{ i === 9 ?
            <section>
              <Link to='/results'><h1>{triviaData}</h1></Link>
            </section>
            : <h1>{triviaData}</h1> }</div>
         </article>
         <article className='test-progress'>
          <h1></h1>
         </article>
       </section>
     )
   }
}

export const mapDispatchToProps = dispatch => ({
  loadTriviaToStore: (trivia) => { dispatch(addTrivia(trivia)) }
})

export const mapStateToProps = state => ({
  trivia: state
});

export default connect(mapStateToProps, mapDispatchToProps)(TestContainer)
