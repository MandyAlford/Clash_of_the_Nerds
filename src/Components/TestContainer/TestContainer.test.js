import React from 'react';
import { TestContainer, mapDispatchToProps, mapStateToProps } from './TestContainer';
import { shallow } from 'enzyme';
import { addTrivia } from '../../actions';


describe('TestContainer', () => {
let wrapper, mockTrivia;

  beforeEach(() => {
    mockTrivia = {trivia: [
        {
          category: 'Mythology',
          type: 'true/false',
          difficulty: 'easy',
          question: 'Loki dressed as a woman in norse mythology',
          correct_answer: 'true',
          incorrect_answers: ['false'],
          userGuess: 'true'
        }, {
          category: 'Mythology',
          type: 'multiple',
          difficulty: 'medium',
          question: 'This Greek mythological figure is the god/goddess of battle strategy (among other things).',
          correct_answer: 'Athena',
          incorrect_answers: ['Ares','Artemis','Apollo'],
          userGuess: 'Athena'
        },{
          category: 'Mythology',
          type: 'multiple',
          difficulty: 'easy',
          question: 'What mytological creatures have women&#039;s faces and vultures&#039; bodies?',
          correct_answer: 'Harpies',
          incorrect_answers: [ 'Mermaids', 'Nymph', 'Lilith' ],
          userGuess: 'Mermaids'
        },{
          category: 'Mythology',
          type: 'boolean',
          difficulty: 'medium',
          question: 'The Roman god &quot;Jupiter&quot; was first known as &quot;Zeus&quot; to the Greeks.',
          correct_answer: 'True',
          incorrect_answers: [ 'False' ],
          userGuess: 'True'
        }]}
      wrapper = shallow(<TestContainer
        trivia={mockTrivia}
        />)
  })


  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should store  chosenAnswer in state when changeCard is invoked', () => {
    let mockTrivia2 = {
          category: 'Mythology',
          type: 'multiple',
          difficulty: 'medium',
          question: 'This Greek mythological figure is the god/goddess of battle strategy (among other things).',
          correct_answer: 'Athena',
          incorrect_answers: ['Ares','Artemis','Apollo'],
          userGuess: 'Athena'
        }
    wrapper.instance().changeCard(mockTrivia2)
    let expected = [{
          category: 'Mythology',
          type: 'multiple',
          difficulty: 'medium',
          question: 'This Greek mythological figure is the god/goddess of battle strategy (among other things).',
          correct_answer: 'Athena',
          incorrect_answers: ['Ares','Artemis','Apollo'],
          userGuess: 'Athena'
        }]
    expect(wrapper.state('chosenAnswer')).toEqual(expected)
  })

  describe('Syntax changes in trivia question data', () => {

    it('should be able to replace &#039; with \'', () => {
      let result = wrapper.instance().formatQuestion()
      let expected = [{
        category: 'Mythology',
        type: 'true/false',
        difficulty: 'easy',
        question: 'Loki dressed as a woman in norse mythology',
        correct_answer: 'true',
        incorrect_answers: ['false'],
        userGuess: 'true'
      },{
        category: 'Mythology',
        type: 'multiple',
        difficulty: 'medium',
        question: 'This Greek mythological figure is the god/goddess of battle strategy (among other things).',
        correct_answer: 'Athena',
        incorrect_answers: ['Ares','Artemis','Apollo'],
        userGuess: 'Athena'
      },{
        category: 'Mythology',
        type: 'multiple',
        difficulty: 'easy',
        question: 'What mytological creatures have women\'s faces and vultures\' bodies?',
        correct_answer: 'Harpies',
        incorrect_answers: [ 'Mermaids', 'Nymph', 'Lilith' ],
        userGuess: 'Mermaids'
      },{
        category: 'Mythology',
        type: 'boolean',
        difficulty: 'medium',
        question: 'The Roman god "Jupiter" was first known as "Zeus" to the Greeks.',
        correct_answer: 'True',
        incorrect_answers: [ 'False' ],
        userGuess: 'True'
      }]
      expect(result).toEqual(expected)
    })

    it('should be able to replace &quot; with "', () => {
      let result = wrapper.instance().formatQuestion()
      let expected = [{
        category: 'Mythology',
        type: 'true/false',
        difficulty: 'easy',
        question: 'Loki dressed as a woman in norse mythology',
        correct_answer: 'true',
        incorrect_answers: ['false'],
        userGuess: 'true'
      },{
        category: 'Mythology',
        type: 'multiple',
        difficulty: 'medium',
        question: 'This Greek mythological figure is the god/goddess of battle strategy (among other things).',
        correct_answer: 'Athena',
        incorrect_answers: ['Ares','Artemis','Apollo'],
        userGuess: 'Athena'
      },{
        category: 'Mythology',
        type: 'multiple',
        difficulty: 'easy',
        question: 'What mytological creatures have women\'s faces and vultures\' bodies?',
        correct_answer: 'Harpies',
        incorrect_answers: [ 'Mermaids', 'Nymph', 'Lilith' ],
        userGuess: 'Mermaids'
      },{
        category: 'Mythology',
        type: 'boolean',
        difficulty: 'medium',
        question: 'The Roman god "Jupiter" was first known as "Zeus" to the Greeks.',
        correct_answer: 'True',
        incorrect_answers: [ 'False' ],
        userGuess: 'True'
      }]
      expect(result).toEqual(expected)
    })
  })

  it('should update state from the store wit trivia data from mapStateToProps', () => {
    let mockState = mockTrivia
    let expected = { trivia: { trivia: [
        {
          category: 'Mythology',
          type: 'true/false',
          difficulty: 'easy',
          question: 'Loki dressed as a woman in norse mythology',
          correct_answer: 'true',
          incorrect_answers: ['false'],
          userGuess: 'true'
        }, {
          category: 'Mythology',
          type: 'multiple',
          difficulty: 'medium',
          question: 'This Greek mythological figure is the god/goddess of battle strategy (among other things).',
          correct_answer: 'Athena',
          incorrect_answers: ['Ares','Artemis','Apollo'],
          userGuess: 'Athena'
        },{
          category: 'Mythology',
          type: 'multiple',
          difficulty: 'easy',
          question: 'What mytological creatures have women&#039;s faces and vultures&#039; bodies?',
          correct_answer: 'Harpies',
          incorrect_answers: [ 'Mermaids', 'Nymph', 'Lilith' ],
          userGuess: 'Mermaids'
        },{
          category: 'Mythology',
          type: 'boolean',
          difficulty: 'medium',
          question: 'The Roman god &quot;Jupiter&quot; was first known as &quot;Zeus&quot; to the Greeks.',
          correct_answer: 'True',
          incorrect_answers: [ 'False' ],
          userGuess: 'True'
        }]}}
    const result = mapStateToProps(mockState)
    expect(result).toEqual(expected)
  })

  it('should  call dispatch with the addTrivia action when loadTriviaToStore is called', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = addTrivia(mockTrivia)
    const mappedProps = mapDispatchToProps(mockDispatch)
    mappedProps.loadTriviaToStore(mockTrivia)
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })

})
