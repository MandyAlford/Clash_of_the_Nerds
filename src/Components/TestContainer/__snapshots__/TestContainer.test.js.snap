// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestContainer should match snapshot 1`] = `
<section>
  <h1
    className="trivia-category"
  >
    Mythology
  </h1>
  <h2
    className="trivia-difficulty"
  >
    Difficulty: 
    easy
  </h2>
  <article
    className="test-question"
  >
    <h1
      className="trivia-header"
    >
      Question: 
      1
       out of 10
    </h1>
    <h1
      className="trivia-question"
    >
      Loki dressed as a woman in norse mythology
    </h1>
  </article>
  <article
    className="test-choice"
  >
    <div>
      <h1>
        <Connect(TriviaCard)
          changeCard={[Function]}
          trivia={
            Object {
              "category": "Mythology",
              "correct_answer": "true",
              "difficulty": "easy",
              "incorrect_answers": Array [
                "false",
              ],
              "question": "Loki dressed as a woman in norse mythology",
              "type": "true/false",
              "userGuess": "true",
            }
          }
        />
      </h1>
    </div>
  </article>
  <article
    className="test-progress"
  >
    <h1 />
  </article>
</section>
`;
